nodejs/routes.js

```diff
@@ -0,0 +1,143 @@
+const express = require('express');
+const router = express.Router();
+const fs = require('fs');
+
+var data = {};
+
+// @todo Do this for all services.
+
+let services = {
+    elasticsearch: 'Elasticsearch',
+    memcached: 'MemcacheD',
+    mongodb: 'MongoDB',
+    mysql: 'MySQL',
+    postgresql: 'PostgreSQL',
+    redis: 'Redis',
+    solr: 'Solr',
+};
+
+Object.keys(services).forEach((key) => {
+    data[key] = require(`./examples/${key}.js`);
+    data[key].source = escapeHtml(fs.readFileSync(`./examples/${key}.js`, 'utf8'));
+    data[key].label = services[key];
+});
+
+
+function escapeHtml(s) {
```

Comment:
Is this where you're handling non-existent service paths? In any case, it's outputting `Error: ENOENT: no such file or directory, stat '/app/examples/something.js'` instead of `Sorry, no sample code is available.`

Reply:
I think this is just escaping html formatting when you're pulling in source, but I can't quite find my way around to where you're handling incorrect paths.

Reply:
Added proper 404 handling, which was the actual bug.